GUI/src/gui_qt.py

```diff
@@ -0,0 +1,458 @@
+                return
+            self.log_message(f"Opening log directory: {path}")
+            subprocess.run(["xdg-open", path], check=True)
+        except subprocess.CalledProcessError as e:
+            self.log_message(f"Error opening log directory {path}: {e}", "error")
+        except Exception as e:
+            self.log_message(f"Unexpected error opening log directory {path}: {e}", "error")
+
+    def browse_folder(self):
+        folder = QFileDialog.getExistingDirectory(self, "Select cdasim-config Repo Folder")
+        if not folder:
+            self.log_message("No folder selected.")
+            return
+        
+        self.repo_folder = folder
+        valid_configs = []
+        
+        for subdir in os.listdir(folder):
+            sub_path = os.path.join(folder, subdir)
+            if os.path.isdir(sub_path):
+                docker_compose = os.path.join(sub_path, "docker-compose.yml")
+                build_sh = os.path.join(sub_path, "build-image.sh")
+                config_dir = os.path.join(sub_path, "cdasim_config")
+                start_sh = os.path.join(config_dir, "start_simulation")
+                stop_sh = os.path.join(config_dir, "stop_simulation")
+                map_dir = os.path.join(config_dir, "MAP")
+                route_dir = os.path.join(config_dir, "route_config")
+                
+                if (os.path.isfile(docker_compose) and
+                    os.path.isfile(build_sh) and
+                    os.path.isdir(config_dir) and
+                    os.path.isfile(start_sh) and
+                    os.path.isfile(stop_sh) and
+                    os.path.isdir(map_dir) and os.listdir(map_dir) and
+                    os.path.isdir(route_dir) and os.listdir(route_dir)):
+                    valid_configs.append(subdir)
+        
+        if not valid_configs:
+            self.log_message("Error: No valid configs found in the repo.", "error")
+            self.config_combo.setEnabled(False)
+            self.config_combo.clear()
+            self.start_button.setEnabled(False)
+            return
+        
+        self.config_combo.blockSignals(True)  # Block signals to prevent automatic trigger
+        self.config_combo.clear()
+        self.config_combo.addItems(sorted(valid_configs))
+        self.config_combo.setCurrentIndex(-1)  # No default selection
+        self.config_combo.blockSignals(False)
+        self.config_combo.setEnabled(True)
+        self.log_message(f"Found {len(valid_configs)} valid config(s): {', '.join(valid_configs)}")
+
+    def on_config_selected(self, config):
+        if not config:
+            self.log_message("No config selected.")
+            return
+        
+        self.selected_folder = os.path.join(self.repo_folder, config)
+        self.log_message(f"Selected config: {config}")
+        
+        self.docker_compose_file = os.path.join(self.selected_folder, "docker-compose.yml")
```

Comment:
Again, not necessary to change for this MVP version but would be nice to validate that these files exist at this level (or before showing them in the selectable list even), rather than when the user attempts to actually use the config.

Reply:
I think the file validation is handled in the `browse_folder` function at line 172, if we're referring to the same functionality. Based on your previous suggestion, when a user selects a `cdasim-config` folder, the system should automatically populate the dropdown list GUI with the validated `cdasim-config` contents. The validated `cdasim-config` must include the following:

1. docker-compose.yml
2. build-image.sh
3. cdasim_config folder
4. Start/stop simulation scripts
5. MAP folder containing .osm files
6. route_config folder containing route .csv files